File: cartipay-main/Pardakht.PardakhtPay/FrontEnd/Pardakht.PardakhtPay.AdminPortal/src/app/models/autotransfer.ts
import { ListSearchArgs, AgGridSearchArgs } from "./types";
import { GenericHelper } from "../helpers/generic";

export class AutoTransfer {
    id: number;

    bankLoginId: number;

    bankAccountId: number;

    tenantGuid: string;

    ownerGuid: string;

    amount: number;

    status: number;

    statusDescription: string;

    requestId: number;

    requestGuid: string;

    cardToCardAccountId: number;

    accountGuid: string;

    transferFromAccount: string;

    transferToAccount: string;

    priority: number;

    transferRequestDateStr: string;

    transferredDateStr: string;

    cancelDateStr: string;

    isCancelled: boolean;

    friendlyName: string;
}

export class AutoTransferSearchArgs extends AgGridSearchArgs {

    tenants: string[];

    dateRange: string;

    constructor(data?: any) {
        super();
        GenericHelper.populateData(this, data);
    }
}

File: cartipay-main/Pardakht.PardakhtPay/BackEnd/Pardakht.PardakhtPay.Shared.Models/Entities/AutoTransfer.cs
using System;
using System.ComponentModel.DataAnnotations;
using Pardakht.PardakhtPay.Shared.Models.Models;

namespace Pardakht.PardakhtPay.Shared.Models.Entities
{
    public class AutoTransfer : BaseEntity, ITenantGuid, IOwnerGuid
    {
        [MaxLength(70)]
        public string TenantGuid { get; set; }

        [MaxLength(70)]
        public string OwnerGuid { get; set; }

        public decimal Amount { get; set; }

        public int Status { get; set; }

        public string StatusDescription { get; set; }

        public int RequestId { get; set; }

        public string RequestGuid { get; set; }

        public int CardToCardAccountId { get; set; }

        [MaxLength(70)]
        public string AccountGuid { get; set; }

        public string TransferFromAccount { get; set; }

        public string TransferToAccount { get; set; }

        public int Priority { get; set; }

        public DateTime TransferRequestDate { get; set; }

        public DateTime? TransferredDate { get; set; }

        public DateTime? CancelDate { get; set; }

        public bool IsCancelled { get; set; }
    }
}
